### ISSUE TITLE
MAUI Community Toolkit Popup Displayed Out Window Bounds On MacCatalyst

### ISSUE THREAD

**Issue by mattleibow** [CommunityToolkit/Maui#898]
---
**Issue moved from dotnet/maui#12652**
- Please respond to @vsfeedback.

---

_From @vsfeedback on Friday, January 13, 2023 7:07:54 PM_

_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/MAUI-Community-Toolkit-Popup-Displayed-O/10191700)._

---
If your app windows is not full screen on Mac when your run your app the popup page won’t show up, now maximize you window and then show the popup and the popup will be shown but it will go ourside the parent window boundaries although I sat the popup page width to the same value as the parent window width (same goes for height of course).

I am attaching two screenshots the first one shows the output when my window is not full screen (vertically in this particular case) which shows only the white translucent background color but the popup is not shown.

Second screenshot shows the popup after I maximized my window vertically to full screen and you can see here that the popup page went out of its parent window from the right side.

---
#### Feedback Bot on 11/6/2022, 11:03 PM: 

(private comment, text removed)
#### Zack Yang [MSFT] on 11/11/2022, 00:22 AM: 

(private comment, text removed)
#### Tamer Irdais on 11/11/2022, 08:34 AM: 

(private comment, text removed)
#### Feedback Bot on 11/13/2022, 11:54 PM: 

(private comment, text removed)
#### Zack Yang [MSFT] on 11/13/2022, 11:58 PM: 

(private comment, text removed)

---
#### Feedback Bot solved on 11/8/2022, 01:07 AM, 0 votes: 

<p>Thanks for taking the time to report this issue! It seems to us that it’s describing the same thing as <a target='_blank' href="https://developercommunity.visualstudio.com/t/MAUI-Community-Toolkit-Popup-Background-/10191178">another one</a>. We’ve closed this one as a duplicate and transferred all votes from this ticket to the other one. Please follow along on the linked ticket to communicate with the team, see updates on status, and help provide any needed diagnostic info. For more information, see our <a target='_blank' href="https://aka.ms/vsfeedbackguidelines">issue reporting guidelines</a>.</p>
<p>Happy coding!</p>

#### Tamer Irdais on 11/9/2022, 07:44 AM: 

(private comment, text removed)

**Comment by mattleibow** [comment]
@TamerIrdais I think this is your other issue?

**Comment by mattleibow** [comment]
This issue has more details here: https://developercommunity.visualstudio.com/t/MAUI-Community-Toolkit-Popup-Displayed-O/10191700

**Comment by mattleibow** [comment]
I submitted this bug from my Mac machine and sent screenshots and demo project but it seems they were not successfully submitted from Mac, so I am adding them again from my Windows machine.

Now, here are the steps to reproduce this bug:

1. Build and run the demo I am attaching with this bug on MacCatalyst.
2. The demo main window will start at the center of the screen with its default window size (demo window not maximized).
3. Now press the “Show Popup” button and you notice there is semi-transparent white layer covering the parent window but the popup itself is NOT shown (see “Screenshot_1.png”) below:


![image](https://user-images.githubusercontent.com/1096616/212772902-3f833aa4-ec51-4fcc-9676-11ad1a31d898.png)


4. Click on the transparent layer to close the popup then resize the demo window vertically so it fills your mac screen vertically (keep the window width as is don’t resize it).
5. Now press the “Show Popup” button again and now you can see the popup page at the bottom of the app window but as you can see it goes outside the parent demo window from the right side of the demo as you can see in “Screenshot_2.png” below and this is the bug I am talking about:

![image](https://user-images.githubusercontent.com/1096616/212772963-e986eb88-7d9c-450a-b288-ac51d7d30d9f.png)


6. Now I also noticed that you can resize the demo’s main window while the popup is still shown which means the popup page is not modal and this behavior is wrong and does NOT happen when you run same demo on WinUI3 (Windows platform).

I hope I made my self clear with these steps, and thank you for your feedback.

[B389524bfec434cc78c182040a3af7f8c638037812804422574_MauiAppNet7.zip](https://github.com/CommunityToolkit/Maui/files/10429707/B389524bfec434cc78c182040a3af7f8c638037812804422574_MauiAppNet7.zip)

**Comment by mattleibow** [comment]
Probably related: https://github.com/CommunityToolkit/Maui/issues/899

**Comment by TamerIrdais** [comment]
> Probably related: #899

@mattleibow yes this is my other issue, and NO its NOT related to #899 this is a totally different issue, the other one is about popup background transparency and this one as you can see in the screenshots I provided is about the popup getting out of the main app window.

**Comment by pictos** [comment]
@TamerIrdais are you specifying a size for your popup? Can you share a link with a sample project?
Friendly reminder that we don't accept zip files, just repos on GitHub or GitLab, more info [here](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug)

**Comment by TamerIrdais** [comment]
@pictos here is the link for the sample demo:

https://github.com/TamerIrdais/MAUI-Community-Toolkit-Popup-Displayed-Out-Window-Bounds-On-MacCatalyst

And yes I am specifying a size for the popup to match the parent window width.

**Comment by TamerIrdais** [comment]
@pictos I didn't hear again from you after I answered your question, is there any update related to this bug?

**Comment by TiDaGo** [comment]
my buttons are off screen )))
and my scroll view not working (

![image](https://user-images.githubusercontent.com/27810072/232403185-85f2dd4d-81e4-4b68-bd0b-31166dd4e81e.png)

<?xml version="1.0" encoding="utf-8" ?>
<toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
                          xmlns:local="clr-namespace:Blog4Van.ViewModels"
           xmlns:cv="clr-namespace:B4V.Components;assembly=B4V.Components"
           xmlns:modalpages="clr-namespace:Blog4Van.Models.ModalPages"
             x:Class="Blog4Van.Views.MSelectPopup">
  <Grid BackgroundColor="#fff"
        RowDefinitions="*,Auto"
        Padding="10"
        VerticalOptions="Fill">
    <ScrollView VerticalOptions="FillAndExpand"
                Grid.Row="0">
      <StackLayout Orientation="Vertical"
                     x:DataType="local:MultiselectViewModel"
                     BindableLayout.ItemsSource="{Binding Items}">
        <BindableLayout.ItemTemplate>
          <DataTemplate>
            <StackLayout x:DataType="modalpages:BaseCheckedItem">
              <Grid Padding="0,0,0,5">
                <Grid.GestureRecognizers>
                  <TapGestureRecognizer Command="{Binding TapCommand}"/>
                </Grid.GestureRecognizers>
                <Grid.RowDefinitions>
                  <RowDefinition  Height="35" />
                </Grid.RowDefinitions>
                <Grid.ColumnDefinitions>
                  <ColumnDefinition Width="35" />
                  <ColumnDefinition Width="*" />
                  <ColumnDefinition Width="50" />
                </Grid.ColumnDefinitions>
                <Image Source="{Binding SvgImageName}"
                               WidthRequest="32"/>
                <Label Text="{Binding Title}"
                         LineBreakMode="NoWrap"
                         VerticalOptions="Center"
                         HorizontalOptions="Start"
                         Grid.Column="1" />
              </Grid>
            </StackLayout>
          </DataTemplate>
        </BindableLayout.ItemTemplate>
      </StackLayout>
    </ScrollView>
    <StackLayout Orientation="Horizontal" HorizontalOptions="CenterAndExpand" Grid.Row="1">
      <Button Text="Apply" Clicked="ApplyButton_Clicked"/>
      <Button Text="Cancel" Clicked="CancelButton_Clicked"/>
    </StackLayout>
  </Grid>
</toolkit:Popup>

**Comment by vhugogarcia** [comment]
The issue has been fixed by using the latest version of the MAUI Community Toolkit 7.0.0 and .NET MAUI 8 in MacCatalyst. Please find attached a couple of screenshots of the popup being within the window using different sizes:

<img width="1027" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/1047398/27d2825c-a0ae-481f-b8e2-850a34ad063b">
<img width="565" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/1047398/a18bf5b0-fc32-4ecb-9ce3-34ffd97fc1a3">

@brminnick I believe we can safely close this issue.
